packages/tools/client-debugger/client-debugger-view/src/Debugger.tsx

```diff
@@ -32,48 +39,68 @@ export interface FluidClientDebuggerProps {
  * @remarks If no debugger has been initialized, will display a note to the user and a refresh button to search again.
  */
 export function FluidClientDebugger(props: FluidClientDebuggerProps): React.ReactElement {
-    /**
-     * This function will retrieve all client debuggers and return the first one.
-     * TODO: update it to check container id after support multi-containers app.
-     */
     function getFirstDebugger(): IFluidClientDebugger | undefined {
-        const clientDebuggers = getFluidClientDebuggers();
-        return clientDebuggers.length === 0 ? undefined : clientDebuggers[0];
+        const debuggers = getFluidClientDebuggers();
+        return debuggers.length === 0 ? undefined : debuggers[0];
     }
 
+    const [clientDebuggers, setClientDebuggers] = React.useState<IFluidClientDebugger[]>(getFluidClientDebuggers());
+
+    const [selectedContainerId, setSelectedContainerId] = React.useState<string>(
+        getFirstDebugger()?.containerId ?? "",
+    );
+
     const [clientDebugger, setClientDebugger] = React.useState<IFluidClientDebugger | undefined>(
-        getFirstDebugger(),
+        selectedContainerId === undefined ? undefined : getFluidClientDebugger(selectedContainerId),
     );
 
     const [isContainerDisposed, setIsContainerDisposed] = React.useState<boolean>(
         clientDebugger?.disposed ?? false,
     );
 
+    const debuggerRegistry: DebuggerRegistry = getDebuggerRegistry();
+
     React.useEffect(() => {
         function onDebuggerDisposed(): void {
             setIsContainerDisposed(true);
         }
 
+        function onDebuggerChanged(): void {
+            setClientDebuggers(getFluidClientDebuggers());
+        }
+
+        debuggerRegistry.on("debuggerRegistered", onDebuggerChanged);
+        debuggerRegistry.on("debuggerClosed", onDebuggerChanged);
         clientDebugger?.on("disposed", onDebuggerDisposed);
 
         return (): void => {
             clientDebugger?.off("disposed", onDebuggerDisposed);
+            debuggerRegistry.off("debuggerRegistered", onDebuggerChanged);
+            debuggerRegistry.off("debuggerClosed", onDebuggerChanged);
         };
-    }, [clientDebugger, setIsContainerDisposed]);
+    }, [clientDebugger, clientDebuggers, setIsContainerDisposed, setClientDebuggers]);
```

Comment:
We need to add `debuggerRegistry` to the list of dependencies here. Otherwise if it changes, this effect won't re-run.

It also doesn't look like we're using `clientDebuggers` anywhere in this effect, so it can probably be removed from the deps list.

Reply:
I see! Fixed it. Thx!

Reply:
Looks like it's still in the dependencies list :)